code/datums/emergency_calls/emergency_call.dm

```diff
@@ -0,0 +1,217 @@
+	candidate_timer = null
+	ticker.mode.waiting_for_candidates = FALSE
+
+	var/list/valid_candidates = list()
+
+	for(var/i in candidates)
+		var/datum/mind/M = i
+		if(!istype(M)) // invalid
+			continue
+		if(M.current) //If they still have a body
+			if(M.current.stat != DEAD) // and not dead or admin ghosting,
+				to_chat(M.current, "<span class='warning'>You didn't get selected to join the distress team because you aren't dead.</span>")
+				continue
+		valid_candidates += M
+
+	message_admins("Distress beacon: [name] got [length(candidates)] candidates, [length(valid_candidates)] of them were valid.")
+
+	if(mob_min && length(valid_candidates) < mob_min)
+		message_admins("Aborting distress beacon [name], not enough candidates. Found: [length(valid_candidates)]. Minimum required: [mob_min].")
+		ticker.mode.waiting_for_candidates = FALSE
+		members.Cut() //Empty the members list.
+		candidates.Cut()
+
+		if(announce)
+			pr_announce.Announce("The distress signal has not received a response, the launch tubes are now recalibrating.", "Distress Beacon")
+
+		ticker.mode.picked_call = null
+		ticker.mode.on_distress_cooldown = TRUE
+
+		cooldown_timer = addtimer(CALLBACK(src, .proc/reset), 5 MINUTES, TIMER_STOPPABLE)
+		return
+
+	var/list/datum/mind/picked_candidates = list()
+	if(length(valid_candidates) > mob_max)
+		for(var/i in 1 to mob_max)
+			if(!length(valid_candidates)) //We ran out of candidates.
+				break
+			picked_candidates += pick_n_take(valid_candidates) //Get a random candidate, then remove it from the candidates list.
+
+		for(var/datum/mind/M in valid_candidates)
+			if(M.current)
+				to_chat(M.current, "<span class='warning'>You didn't get selected to join the distress team. Better luck next time!</span>")
+		message_admins("Distress beacon: [length(valid_candidates)] valid candidates were not selected.")
+	else
+		picked_candidates = valid_candidates // save some time
+		message_admins("Distress beacon: All valid candidates were selected.")
+
+	if(announce)
+		pr_announce.Announce(dispatch_message, "Distress Beacon")
+
+	message_admins("Distress beacon: [name] finalized, starting spawns.")
+
+	if(mob_min > 0)
+		if(length(picked_candidates))
+			for(var/i in picked_candidates)
+				var/datum/mind/candidate_mind = i
+				members += candidate_mind
+				create_member(candidate_mind)
+		else
+			message_admins("ERROR: No picked candidates, aborting.")
+			return
+
+	message_admins("Distress beacon: [name] finished spawning.")
+
+	candidates.Cut() //Blank out the candidates list for next time.
+
+	cooldown_timer = addtimer(CALLBACK(src, .proc/reset), 1 MINUTES, TIMER_STOPPABLE)
+
+/datum/emergency_call/proc/add_candidate(mob/M)
+	if(!M.client)
+		return FALSE  //Not connected
+
+	if(M.mind && (M.mind in candidates))
+		return FALSE  //Already there.
+
+	if(M.stat != DEAD)
+		return FALSE  //Alive, could have been drafted into xenos or something else.
+
+	if(!M.mind) //They don't have a mind
```

Comment:
don't check mind twice, do this one first

Reply:
also, use get_mind instead of .mind

Reply:
This check mind. Because M.mind && (M.mind in candidates) can TRUE with mind = null.

Reply:
check mind once, after client.

then the second check if mind is in candidates